robot-server/tests/protocols/test_protocol_store.py

```diff
@@ -13,23 +13,26 @@
     PythonProtocolConfig,
 )
 
-from robot_server.db import opened_db
 from robot_server.protocols.protocol_store import (
-    add_tables_to_db,
     ProtocolStore,
     ProtocolResource,
     ProtocolNotFoundError,
 )
+from robot_server.persistence import open_db_no_cleanup, add_tables_to_db
 
 from sqlalchemy.engine import Engine as SQLEngine
 
 
 @pytest.fixture
 def sql_engine(tmp_path: Path) -> Generator[SQLEngine, None, None]:
     """Return a set-up database to back the store."""
-    with opened_db(db_file_path=tmp_path / "test.db") as engine:
-        add_tables_to_db(engine)
-        yield engine
+    db_file_path = tmp_path / "test.db"
+    sql_engine = open_db_no_cleanup(db_file_path=db_file_path)
+    add_tables_to_db(sql_engine)
+    try:
+        yield sql_engine
+    finally:
+        sql_engine.dispose()
```

Comment:
Minor tweak:

```suggestion
    db_file_path = tmp_path / "test.db"
    sql_engine = open_db_no_cleanup(db_file_path=db_file_path)
    try:
        add_tables_to_db(sql_engine)
        yield sql_engine
    finally:
        sql_engine.dispose()
```

Because if `add_tables_to_db()` raises an exception, `sql_engine` still needs to be cleaned up.